index.bs

```diff
@@ -1,11 +1,62 @@
-<pre class='metadata'>
-Title: Web Extensions
-Shortname: webextensions
-Level: 1
-Status: w3c/CG-DRAFT
-Group: WECG
-URL: https://w3c.github.io/webextensions
-Editor: Mukul Purohit, Microsoft Corporation https://www.microsoft.com, mpurohit@microsoft.com
-Editor: Tomislav Jovanovic, Mozilla https://www.mozilla.org/, tjovanovic@mozilla.com
-Abstract: [Placeholder] Abstract.
-</pre>
+<pre class='metadata'>
+Title: Web Extensions
+Shortname: webextensions
+Level: 1
+Status: w3c/CG-DRAFT
+Group: WECG
+URL: https://w3c.github.io/webextensions
+Editor: Mukul Purohit, Microsoft Corporation https://www.microsoft.com, mpurohit@microsoft.com
+Editor: Tomislav Jovanovic, Mozilla https://www.mozilla.org/, tjovanovic@mozilla.com
+Abstract: [Placeholder] Abstract.
+Markup Shorthands: markdown yes
+</pre>
+
+# Manifest
+
+A WebExtension must have a manifest file at its root directory.
+
+## Manifest file
+
+A manifest file is a [[!JSON]] document named `manifest.json`. Malformed JSON files are not supported.
+
+## Manifest keys
+
+If manifest keys that are not defined in this specification are specified, implementors may ignore those keys.
+
+If manifest keys that are defined in this specification are specified with a different JSON type than defined in this specification, implementors may ignore those keys.
+
+A manifest file must have these keys properly specified:
+
+* <a href="#key-name">`name`</a>: required.
+* <a href="#key-version">`version`</a>: required.
+* <a href="#key-default_locale">`default_locale`</a>: required under some conditions.
+* <a href="#key-description">`description`</a>: optional.
+* <a href="#key-icons">`description`</a>: optional.
+
+### Key `name`
+
+This key must be present. Its value must be a JSON string.
+
+It should be used as the user-facing name of the extension.
+
+Note: TODO explain translations in `manifest.json`.
+
+### Key `version`
+
+This key must be present. It must be a JSON string.
+
+Note: TODO describe how versions should be compared
+
+### Key `default_locale`
+
+This key must be present if the `_locales` subdirectory is present, must be absent otherwise. Its value must be a JSON string.
+
+### Key `description`
+
+This key may be present. Its value must be a JSON string.
+
+It should be used.
+
+### Key `icons`
+
+This key may be present. It value must be a JSON object, whose keys and values are JSON strings.
```

Comment:
Should we use JSON Schema to specify those things?

Reply:
I think we should but later probably, since we don’t have a reference text yet.

Reply:
Wouldn't the JSON schema for the manifest be a separate document, and this would be more prose?

Reply:
@xeenon, I'm actually unclear about that at the moment. I've seen references in the Web Application Manifest issues to parsing content out of the spec file. I'm currently reaching out to Googlers that have contributed to the Web Application Manifest spec to gather information on how that WG has approached JSON specification and integration with browser tooling.

Reply:
Could you open an issue about this? I believe the question is if we can use a JSON schema file as part of the spec so the «core» text is lighter, right?

Reply:
Will do.